python/ray/train/xgboost/config.py

```diff
@@ -95,7 +166,16 @@ def set_xgboost_env_vars():
 
         worker_group.execute(set_xgboost_env_vars)
 
+    def on_training_start(
+        self, worker_group: WorkerGroup, backend_config: XGBoostConfig
+    ):
+        assert backend_config.xgboost_communicator == "rabit"
```

Comment:
nit: it seems `XGBoostingConfig` has a hard coded `backend_config` field being `"rabit"`, why do we still need an assertion here?

Reply:
yeah, I can probably remove this field for now, since we don't support the "federated" option.